v-next/hardhat/test/internal/builtin-plugins/network-manager/type-validation.ts

```diff
@@ -0,0 +1,396 @@
+            "Expected 'remote', an array of strings, or an object with account details",
+          );
+        }
+      });
+
+      it("should fail with invalid HttpNetworkHDAccountsConfig", () => {
+        // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- the wrong type must be forced to trigger the test failure
+        const accountsValuesToTest = [
+          { mnemonic: 123 },
+          { initialIndex: "asd" },
+          { count: "asd" },
+          { path: 123 },
+          { type: 123 },
+          {
+            mnemonic: "hello",
+            initialIndex: 1,
+            count: 1,
+            path: "path",
+          },
+        ] as any;
+
+        for (const accounts of accountsValuesToTest) {
+          httpNetworkConfig.accounts = accounts;
+
+          const validationErrors = validateNetworkConfig(httpNetworkConfig);
+
+          assert.notEqual(validationErrors.length, 0);
+          assert.equal(
+            validationErrors[0].message,
+            "Expected 'remote', an array of strings, or an object with account details",
+          );
+        }
+      });
+    });
+  });
+});
+
+describe("EdrNetworkConfig", function () {
+  describe("accounts", function () {
+    const edrNetworkConfig: EdrNetworkConfig = {
+      type: "edr",
+      chainId: 1,
+      from: "",
+      gas: "auto",
+      gasMultiplier: 1,
+      gasPrice: "auto",
+      accounts: [
+        // Modified in the tests
+        {
+          privateKey: "",
+          balance: "0x0",
+        },
+      ],
+    };
+
+    describe("allowed values", () => {
+      it("should allow an array of account objects with valid private keys", function () {
+        edrNetworkConfig.accounts = [
+          {
+            balance: "123",
+            privateKey:
+              "0xaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+          },
+          {
+            balance: "123",
+            privateKey:
+              "0xbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb",
+          },
+          {
+            balance: "123",
+            privateKey:
+              "0xcccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc",
+          },
+        ];
+
+        const validationErrors = validateNetworkConfig(edrNetworkConfig);
+
+        assert.equal(validationErrors.length, 0);
+      });
+
+      it("should allow an account with a valid EdrNetworkHDAccountsConfig", function () {
+        edrNetworkConfig.accounts = {
+          mnemonic: "asd asd asd",
+          initialIndex: 0,
+          count: 123,
+          path: "m/1/2",
+          accountsBalance: "123",
+          passphrase: "passphrase",
+        };
+
+        const validationErrors = validateNetworkConfig(edrNetworkConfig);
+
+        assert.equal(validationErrors.length, 0);
+      });
+    });
+
+    describe("not allowed values", () => {
+      describe("wrong private key formats", () => {
+        // TODO: still valid?
```

Comment:
Same comment as above:
 
 Do we still wanna keep the possibility of allowing valid private keys with missing hex prefix?
If yes, I will have to modify the hh-utils method that I added to check if an address is valid, [see here](https://github.com/NomicFoundation/hardhat/pull/5817/files#diff-8d9a1ba6601820ff6249438ad3797d59980861eef0ca459e2a05512362894bcd)